### PR TITLE
PPC Video Optimization: Ignore local files

### LINKED ISSUE #7472: Video Optimization PPC: Respect `local` attribute when loading/optimizing
`Video not Optimized` tip should not show while `element.resource.local === true` . this prevents the check from prematurely showing while optimization is in progress

### PR DESCRIPTION
**Pull request by maxyinger**
## Context
Fine tuning video optimization check in the PPC.

## Summary
Opts out of the ppc video needs optimization check if `resource.local` is true which indicates that the resource is loading or being optimized.

Also removed check for `isCoverr` which @swissspidy requested be done somewhere else.

## Relevant Technical Choices
NA

## To-do
NA

## User-facing changes
NA

## Testing Instructions
Open editor, make sure auto optimization is enabled, drag large video from your desktop directly onto the story canvas. See that while the video is optimizing, the PPC check to indicate `video not optimized` doesn't show up.

### QA
NA

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7472

### CODE REVIEW

**Review by embarks — APPROVED** [review]
I will use this 👏

**Review by BrittanyIRL — APPROVED** [review]
👏  👏 👏

**Review by spacedmonkey — DISMISSED** [review]
See comment - https://github.com/google/web-stories-wp/issues/7472#issuecomment-838162355

**Comment by spacedmonkey** [comment]
Please add `isOptimized: true` here.
https://github.com/google/web-stories-wp/blob/ee98d46ddfec1c1224ca450980e3d896aa8c4678/assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js#L297

**Review by samwhale — APPROVED** [review]

**Comment by swissspidy** [comment]
> Please add `isOptimized: true` here.
> https://github.com/google/web-stories-wp/blob/ee98d46ddfec1c1224ca450980e3d896aa8c4678/assets/src/edit-story/app/media/utils/getResourceFromMedia3p.js#L297

@spacedmonkey @littlemilkstudio This is best done in a new ticket & PR as it ideally also has a data migration. See discussion at #7472

**Comment by spacedmonkey** [comment]
> @spacedmonkey @littlemilkstudio This is best done in a new ticket & PR as it ideally also has a data migration. See discussion at #7472 

Okay, but this should be done in another ticket, merged. It should block this ticket. Until we make that change, all coverr videos would trigger PPC.

File: assets/src/edit-story/app/prepublish/guidance/media.js [hunk 1]
```diff
@@ -145,15 +145,16 @@ export function videoElementLength(element) {
...
     (element.resource?.height ?? 0) * (element.resource?.width ?? 0);
   const isLargeVideo =
     videoArea >= VIDEO_SIZE_THRESHOLD.WIDTH * VIDEO_SIZE_THRESHOLD.HEIGHT;
 
-  if (!isCoverrMedia && isLargeVideo && !element.resource?.isOptimized) {
+  if (
+    isLargeVideo &&
+    !element.resource?.local &&
+    !element.resource?.isOptimized
+  ) {
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/guidance/media.js]: I'd just write:

```js
if (element.resource?.local) {
  return undefined;
}

if (element.resource?.isOptimized) {
  return undefined;
}

if (isLargeVideo) {
 return {...}
}
```

**Comment by swissspidy** [comment]
> > @spacedmonkey @littlemilkstudio This is best done in a new ticket & PR as it ideally also has a data migration. See discussion at #7472
> 
> Okay, but this should be done in another ticket, merged. It should block this ticket. Until we make that change, all coverr videos would trigger PPC.

The current `isCoverrMedia` check can be kept in this PR and then we fix it in another PR.

File: assets/src/edit-story/app/prepublish/guidance/media.js [hunk 2]
```diff
@@ -145,15 +145,16 @@ export function videoElementLength(element) {
  * @return {Guidance|undefined} The guidance object for consumption
  */
 export function videoElementOptimized(element = {}) {
-  const idResource = element.resource?.id;
-  const idOrigin = idResource?.toString().split(':')?.[0];
-  const isCoverrMedia = idOrigin === 'media/coverr';
```
Comment (swissspidy) [inline assets/src/edit-story/app/prepublish/guidance/media.js:150]: Let's keep this here. This can be changed in another PR.

File: assets/src/edit-story/app/prepublish/guidance/media.js (hunk 1 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/prepublish/guidance/media.js]: Why give there more execution paths and early returns. I think this makes the code a little overly verbose

File: assets/src/edit-story/app/prepublish/guidance/media.js (hunk 2 above)
Comment (spacedmonkey) [reply assets/src/edit-story/app/prepublish/guidance/media.js:150]: Agreed, this should be done in a follow up PR.

File: assets/src/edit-story/app/prepublish/guidance/media.js (hunk 1 above)
Comment (swissspidy) [reply assets/src/edit-story/app/prepublish/guidance/media.js]: To reduce complexity / the number of execution paths and make it more readable.

Comment (spacedmonkey) [reply assets/src/edit-story/app/prepublish/guidance/media.js]: I don't personally find this line that hard to read...

Comment (maxyinger) [reply assets/src/edit-story/app/prepublish/guidance/media.js]: I like it. Keeps the conditional from getting to bloated and ugly as we add more checks to it. agree that rn with the amount of checks there aren't as many benefits but 🤷

**Review by swissspidy — APPROVED** [review]

**Review by spacedmonkey — APPROVED** [review]
👍

**Comment by swissspidy** [comment]
Opened #7529 for the follow-up task.
